opensrp-reveal/src/main/java/org/smartregister/reveal/presenter/TaskRegisterFragmentPresenter.java

```diff
@@ -0,0 +1,218 @@
+
+    private PreferencesUtil prefsUtil;
+    private RevealMappingHelper mappingHelper;
+
+    public TaskRegisterFragmentPresenter(TaskRegisterFragmentContract.View view, String viewConfigurationIdentifier) {
+        this(view, viewConfigurationIdentifier, null, new LocationUtils(view.getContext()));
+        this.interactor = new TaskRegisterFragmentInteractor(this);
+    }
+
+    @VisibleForTesting
+    protected TaskRegisterFragmentPresenter(TaskRegisterFragmentContract.View view, String viewConfigurationIdentifier,
+                                            TaskRegisterFragmentInteractor interactor, LocationUtils locationUtils) {
+        this.view = new WeakReference<>(view);
+        this.viewConfigurationIdentifier = viewConfigurationIdentifier;
+        this.interactor = interactor;
+        this.locationUtils = locationUtils;
+        viewsHelper = ConfigurableViewsLibrary.getInstance().getConfigurableViewsHelper();
+        prefsUtil = PreferencesUtil.getInstance();
+        mappingHelper = new RevealMappingHelper();
+
+    }
+
+
+    @Override
+    public void processViewConfigurations() {
+        if (!StringUtils.isBlank(this.viewConfigurationIdentifier)) {
+            ViewConfiguration viewConfiguration = viewsHelper.getViewConfiguration(this.viewConfigurationIdentifier);
+            if (viewConfiguration != null) {
+                visibleColumns = viewsHelper.getRegisterActiveColumns(this.viewConfigurationIdentifier);
+            }
+        }
+    }
+
+    @Override
+    public void initializeQueries(String mainCondition) {
+
+        getView().initializeAdapter(visibleColumns);
+        lastLocation = locationUtils.getLastLocation();
+        if (lastLocation == null) {//if location client has not initialized use last location passed from map
+            lastLocation = getView().getLastLocation();
+        }
+
+        getView().showProgressView();
+
+        interactor.findTasks(getMainCondition(), lastLocation, getOperationalAreaCenter());
+
+    }
+
+    private android.location.Location getOperationalAreaCenter() {
+        Location operationalAreaLocation = Utils.getOperationalAreaLocation(prefsUtil.getCurrentOperationalArea());
+        if (operationalAreaLocation == null)
+            return null;
+        Gson gson = new GsonBuilder().setDateFormat("yyyy-MM-dd'T'HH:mm:ss.SSSZ")
+                .registerTypeAdapter(DateTime.class, new DateTimeTypeConverter()).create();
+        return mappingHelper.getCenter(gson.toJson(operationalAreaLocation.getGeometry()));
+    }
+
+    @Override
+    public void startSync() {
+        Utils.startImmediateSync();
+    }
+
+    @Override
+    public void searchGlobally(String uniqueId) {//do nothing, tasks not searchable globally
+    }
+
+    /**
+     * Gets the where clause for the task register, filters by operational area and campaign
+     *
+     * @return pair of filter clause and values for filter
+     */
+    private Pair<String, String[]> getMainCondition() {
```

Comment:
Why not implement this in the TaskRegisterFragment where it's actually an abstract method that needs implementation?

Reply:
This because am following MVP where logic should not be in views

Reply:
Then there is a problem with this method being an abstract method in a view base class.

It will always have logic else the implementer will have to stub it out, which is inconvenient.